packages/flutter/lib/src/rendering/stack.dart

```diff
@@ -595,11 +599,34 @@ class RenderStack extends RenderBox
 
   /// Override in subclasses to customize how the stack paints.
   ///
-  /// By default, the stack uses [defaultPaint]. This function is called by
-  /// [paint] after potentially applying a clip to contain visual overflow.
+  /// This function is called by [paint] after potentially applying
+  /// a clip to contain visual overflow.
+  /// children are sorted by child's zIndex in an ascending order then painted
```

Comment:
nit: doc comments should be full and grammatically correct sentences starting with a capital letter and ending with a `.`.